### PR TITLE
Hometask1

### PR DESCRIPTION
**Pull request by FercanSen**

### CODE REVIEW

File: src/components/SearchForm.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {posts.map((post) => (
+          <li key={post.id}>{post.name}</li>
+        ))}
+      </ul>
+      <div>Search Result: </div>
+      <ul>
+        {filteredPosts.map((post) => (
+          <li key={post.key}>{post.name}</li>
+        ))}
+      </ul>
```
Comment (inperegelion) [inline src/components/SearchForm.jsx:42]: Good to see that you are using keys, but, I'm afraid all keys for all posts will be undefined :(

File: src/components/SearchForm.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const filterPosts = (posts, query) => {
+    if (!query) {
+      return posts;
+    }
+
+    return posts.filter((post) => {
+      const postName = post.name;
+      return postName.includes(query);
+    });
+  };
```
Comment (inperegelion) [inline src/components/SearchForm.jsx:23]: In React projects it is convenient to move all functions like this to the `src/utils` folder.

File: src/components/SearchForm.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+// import { useState, useEffect } from "react";
```
Comment (inperegelion) [inline src/components/SearchForm.jsx:2]: Do not forget to clean `//comments` when sending on review. Some time ago it was my constant mistake

File: src/components/SearchForm.jsx (hunk 2 above)
Comment (FercanSen) [reply src/components/SearchForm.jsx:23]: Added changes

File: src/components/SearchForm.jsx (hunk 3 above)
Comment (FercanSen) [reply src/components/SearchForm.jsx:2]: Added changes

File: src/components/SearchForm.jsx (hunk 1 above)
Comment (FercanSen) [reply src/components/SearchForm.jsx:42]: Added changes

**Pull request merged**